pkg/gce-cloud-provider/metadata/metadata.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metadata
+
+import (
+	"fmt"
+
+	"cloud.google.com/go/compute/metadata"
+)
+
+type MetadataService interface {
```

Comment:
Comments, especially around the intended purpose of the interface in order to limit its unwanted expansion. I'm worried about future developers adding arbitrary methods here.

nit: "Service" gives me an impression of a long running process, and it's already an overloaded term in the container world